File: src/main/java/com/baraseck/supportflow/service/CommentService.java
package com.baraseck.supportflow.service;

import com.baraseck.supportflow.dto.AddCommentRequest;
import com.baraseck.supportflow.dto.CommentResponse;

import java.util.List;
import java.util.UUID;

public interface CommentService {
    CommentResponse addComment(UUID ticketId, AddCommentRequest request);
    List<CommentResponse> getComments(UUID ticketId);
}

File: frontend/src/app/core/services/comment.service.ts
import { HttpClient } from '@angular/common/http';
import { inject, Injectable } from '@angular/core';
import { environment } from '../../../environments/environment';
import { AddCommentRequest, CommentResponse } from '../models/comment.models';

@Injectable({ providedIn: 'root' })
export class CommentService {
  private readonly http = inject(HttpClient);

  getComments(ticketId: string) {
    return this.http.get<CommentResponse[]>(`${environment.apiBaseUrl}/api/tickets/${ticketId}/comments`);
  }

  addComment(ticketId: string, request: AddCommentRequest) {
    return this.http.post<CommentResponse>(`${environment.apiBaseUrl}/api/tickets/${ticketId}/comments`, request);
  }
}
